### PR TITLE
web: show fatal error as a modal in web UI

### PR DESCRIPTION
**Pull request by jazzdan**
Looks like this:

Screen Shot 2019-09-26 at 11 28 12 AM

### CODE REVIEW

File: web/src/types.ts [hunk 1]
```diff
@@ -103,3 +103,9 @@ export type Snapshot = {
   SnapshotLink: string
   showSnapshotModal: boolean
 }
+
+export enum ShowFatalErrorModal {
+  Default,
+  Show,
```
Comment (jazzdan) [inline web/src/types.ts]: This value isn't used yet but I'd like, in the future, to provide a way to show the Fatal Error Modal even if it has been dismissed, if there is a fatal error.

File: internal/hud/webview/view.go [hunk 2]
```diff
@@ -180,6 +180,8 @@ type View struct {
 
 	TiltCloudUsername   string
 	TiltCloudSchemeHost string
+
+	FatalError string
```
Comment (nicks) [inline internal/hud/webview/view.go]: i wonder if we should start being better about using omitempty

File: web/src/FatalErrorModal.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+type props = {
+  error: string | null
+  showFatalErrorModal: ShowFatalErrorModal
+  handleClose: () => void
+}
+
+export default class FatalErrorModal extends PureComponent<props> {
+  render() {
+    let showModal =
+      this.props.error !== null &&
```
Comment (nicks) [inline web/src/FatalErrorModal.tsx]: can we just use `this.props.error &&`?

i feel like we've had a lot of bugs due to bad use of strict equality. for example, in this case, `""` and `undefined` should not show the modal

File: internal/hud/webview/view.go (hunk 2 above)
Comment (jazzdan) [reply internal/hud/webview/view.go]: Oh interesting! Why?

Comment (nicks) [reply internal/hud/webview/view.go]: 1) reduce payload size
2) reduce diffs when the snapshot json changes

**Pull request merged**